### PR TITLE
Push all fork changes to the base repository

### LINKED ISSUE #36: Implement drag n drop
Try to build vliw code from superescalar code using react dnd. Try to respect all the titles and implement the most simple version using the webapi.

### PR DESCRIPTION
**Pull request by oxcabe**
## Description
This PR pushes all changes from the [oxcabe/SIMDE](https://github.com/oxcabe/SIMDE) fork to the root repository (this one).

The expected date range for **rebasing** these changes from this PR into the `main` branch is set for the last week of July. Until that date, this PR message will be updated with additional information about what has been done, in order to provide better context for the changes.

Fixes #36.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,56 @@
...
+# Copy application dependency manifests to the container image.
+# A wildcard is used to ensure copying both package.json AND package-lock.json (when available).
+# Copying this first prevents re-running npm install on every code change.
+COPY --chown=node:node package*.json ./
+
+# Install app dependencies using the `npm ci` command instead of `npm install`
+RUN npm ci
+
+# Bundle app source
+COPY --chown=node:node . .
```
Comment (adrianabreu) [inline Dockerfile:18]: When setting up I see the  node_modules belonging to the root user still, does this work for you?

**Review by adrianabreu — APPROVED** [review]
Just added a small comment on the dockerfile part, good work!

File: Dockerfile (hunk 1 above)
Comment (oxcabe) [reply Dockerfile:18]: I'll merge this PR now, but this is definitely something that needs review. Thus, I decided to create an issue: https://github.com/etsiiull/SIMDE/issues/68

**Comment by oxcabe** [comment]
It's finally done!

**Pull request merged**